### DISCUSSION TITLE
GoTrue: Custom access token hook: Include current OAuth provider type

### DISCUSSION THREAD

**Discussion by bnjmnt4n** [supabase/supabase#21935] [Feature Requests]
I'd like to use the new Auth Hooks to customize some aspects of the login behavior based on the OAuth provider type used to sign in, via the [custom access token hook](https://supabase.com/docs/guides/auth/auth-hooks#hook-custom-access-token). However, the access token hook doesn't provide any details about the current OAuth provider.

**Comment by danieldickison** [comment]
The [docs](https://supabase.com/docs/guides/auth/auth-hooks/custom-access-token-hook) claim there's a `client_id` field in the claims, so you might be able to use that:
```
"client_id": "oauth-client-id-if-oauth-flow"
```

What we ended up using was querying `auth.identities` for the most recently signed-in provider, like:

```sql
  IF v_auth_method = 'oauth' THEN
    SELECT i.provider INTO v_oauth_provider
    FROM auth.identities i
    WHERE i.user_id = v_user_id
      AND i.provider IN ('google', 'apple')
    ORDER BY i.last_sign_in_at DESC NULLS LAST
    LIMIT 1;
  END IF;
```

But this would definitely be nice to have in the hook payload directly.
